### PR TITLE
15 make the tests work

### LINKED ISSUE #15: make the tests work
Depends on #12

### PR DESCRIPTION
**Pull request by bbean23**
closes #15 

- most of changed are help ensure that the tests pass more reliably
- the changes to requirements.txt are targeting scipy, to ensure that test_photogrammetry.py passes
- the changes to sensitive_strings.py let me update the sensitive_strings_allowed_binary_files.csv more easily to keep up with the reorganization of sofast

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -9,30 +9,30 @@
...
-python-pptx >= 0.6.21
-rawpy >= 0.18.1
+pytest >= 8.1.1
+python-pptx >= 0.6.23
+rawpy >= 0.19.1
 openpyxl >= 3.1.2
 
 # these are packages that have errors with both conda and pip on solo #errpkgs
-pypylon >= 1.9.0
+pypylon >= 3.0.1
```
Comment (e10harvey) [inline requirements.txt:32]:
```suggestion
pypylon >= 2.3.0
```

**Review by e10harvey — CHANGES_REQUESTED** [review]
Thank you, @bbean23! In general everything looks great! Braden and I will be working on the ubi8 test failures tomorrow. Other than the pypylon version change, I think this can be merged as long as no new test failures are introduced from the dependency version changes in requirements.txt.

**Comment by e10harvey** [comment]
> Thank you, @bbean23! In general everything looks great! Braden and I will be working on the ubi8 test failures tomorrow. Other than the pypylon version change, I think this can be merged as long as no new test failures are introduced from the dependency version changes in requirements.txt.

ubi8 camera calibration errors fixed via: #42.

File: requirements.txt (hunk 1 above)
Comment (bbean23) [reply requirements.txt]: I changed the version number and added a warning for when the version is below 3.0

Comment (braden6521) [reply requirements.txt]: Great, thanks for fixing this!

**Review by braden6521 — APPROVED** [review]
All looks good to me, Ben!

**Review by e10harvey — APPROVED** [review]
It looks like there are also some features added to sensitive strings in this PR.

Going forward, I would prefer that features be separated into a different PR from the test fixes.

**Comment by bbean23** [comment]
> Going forward, I would prefer that features be separated into a different PR from the test fixes.

Yeah, I really should have split that out into another ticket, sorry.

Thanks for the review!

**Pull request merged**